spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSparkSqlExtensionsParser.scala

```diff
@@ -41,7 +43,17 @@ import org.apache.spark.sql.types.StructType
 
 class IcebergSparkSqlExtensionsParser(delegate: ParserInterface) extends ParserInterface {
 
-  private lazy val substitutor = new VariableSubstitution(SQLConf.get)
+  private lazy val substitutorCtor: DynConstructors.Ctor[VariableSubstitution] =
```

Comment:
Can you move this to a companion object so it doesn't happen for every instance of the extensions?

Reply:
Done.